core/camel-main/src/main/java/org/apache/camel/main/BaseMainSupport.java

```diff
@@ -1082,4 +1050,69 @@ public int hashCode() {
             return Objects.hash(key, instance);
         }
     }
+
+    public static void computeProperties(String keyPrefix, String key, Properties prop, Map<PropertyOptionKey, Map<String, Object>> properties, Function<String, Object> supplier) {
+        if (key.startsWith(keyPrefix)) {
+            // grab name
+            final int dot = key.indexOf(".", keyPrefix.length());
+            final String name = dot == -1 ? key.substring(keyPrefix.length()) : key.substring(keyPrefix.length(), dot);
+
+            // enabled is a virtual property
+            if ("enabled".equals(name)) {
+                return;
+            }
+            // skip properties as its already keyPrefix earlier
+            if ("properties".equals(name)) {
+                return;
+            }
+
+            // determine if the service is enabled or not by taking into account two options:
+            //
+            //   1. ${keyPrefix}.enabled = true|false
+            //   2. ${keyPrefix}.${name}.enabled = true|false
+            //
+            // The option [2] has the higher priority so as example:
+            //
+            //   camel.component.enabled = false
+            //   camel.component.seda.enabled = true
+            //
+            // enables auto configuration of the seda component only
```

Comment:
Seems code comments is talking about seda

Reply:
This is expected as it is an example, should I change it ?

Reply:
Ah yeah now i see it, yeah this is good as-is